Full docs build shortcuts, checking in snippets (#1760)
temporalio/documentation: CONTRIBUTING.md

To preview snipsync snippets, run `yarn snipsync`, which inserts the snippet contents (in this case, from [`samples-typescript`](https://github.com/temporalio/samples-typescript/blob/75bdcd613bd24f8f357cb96d1b83051353c5685a/hello-world/src/client.ts#L1)) between the `SNIPSTART` and `SNIPEND` tags.

Run `yarn snipsync --clear` to remove the snippets.

We are now committing the snippets into the docs source files.

> See the [full build command](#full-build-command) section for details on how to fully assemble the docs and, format them, and check for build issues.

Reviewer: ```suggestion
See the [full build command](#full-build-command) section for details on how to fully assemble the docs, format them, and check for build issues.
```

3. In another terminal, from the root of the repo, run the following command:

```
yarn assemble --cloud
```

## How to use the full build command

Run `yarn gen` or `yarn gen --cloud`.

> This will run Snipsync, run the Assembly Workflow, run dPrint formatting, and then build the site.

Reviewer: ```suggestion
This command runs snipsync, runs the Assembly Workflow, runs dPrint formatting, and then builds the site.
```
Earlier, you lowercased "snipsync", so I went with that style here.

> Any build issues logged to your console.

Reviewer: ```suggestion
Any build issues are logged to your console.
```
